Link database migration documentation from Galaxy docs

jdavcs opened the pull request:
Follow-up to review of #14537.

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [x] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

mvdbeek commented on doc/source/dev/faq.rst:
```diff
@@ -13,6 +13,12 @@ This can be done with either IPython or a plain Python console, depending on you
+... upgrade, downgrade, or modify the Galaxy database schema?
+-------------------------------------------------------------
+
+Please read Galaxy's [database schema migration documentation](https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/model/migrations/README.md).
```
Can you please link the document itself up ? In the admin section.

jdavcs committed:
Display 'Config Options' as subheading of 'Galaxy Configuration'

This corrects a (supposedly) typo that prevented this to be displayed as
a subheading on this page: https://docs.galaxyproject.org/en/master/admin/index.html

jdavcs committed:
Add db migration documentation to docs
